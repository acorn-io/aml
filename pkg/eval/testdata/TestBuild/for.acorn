for x in list {
	a: 1
}
for i, y in list2 {
	b: for y in [1,2] {
		true,
	}
}
