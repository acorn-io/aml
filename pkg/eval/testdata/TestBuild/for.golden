&eval.File{Body: &eval.Struct{
	Position: eval.Position{
		Filename: "for.acorn",
		Line:     1,
		Column:   1,
	},
	Fields: []eval.Field{
		&eval.Embedded{
			Pos: eval.Position{
				Filename: "for.acorn",
				Line:     1,
				Column:   1,
			},
			Expression: &eval.For{
				Value: "x",
				Collection: &eval.Lookup{
					Pos: eval.Position{
						Filename: "for.acorn",
						Offset:   9,
						Line:     1,
						Column:   10,
					},
					Key: "list",
				},
				Body: &eval.Struct{
					Position: eval.Position{
						Filename: "for.acorn",
						Offset:   14,
						Line:     1,
						Column:   15,
					},
					Fields: []eval.Field{&eval.KeyValue{
						Key: eval.FieldKey{
							Key: eval.Value{
								Value: value.String("a"),
							},
							Pos: eval.Position{
								Filename: "for.acorn",
								Offset:   17,
								Line:     2,
								Column:   2,
							},
						},
						Value: eval.Value{Value: value.Number("1")},
						Pos: eval.Position{
							Filename: "for.acorn",
							Offset:   17,
							Line:     2,
							Column:   2,
						},
					}},
				},
				Merge: true,
				Position: eval.Position{
					Filename: "for.acorn",
					Offset:   4,
					Line:     1,
					Column:   5,
				},
			},
		},
		&eval.Embedded{
			Pos: eval.Position{
				Filename: "for.acorn",
				Offset:   24,
				Line:     4,
				Column:   1,
			},
			Expression: &eval.For{
				Key:   "i",
				Value: "y",
				Collection: &eval.Lookup{
					Pos: eval.Position{
						Filename: "for.acorn",
						Offset:   36,
						Line:     4,
						Column:   13,
					},
					Key: "list2",
				},
				Body: &eval.Struct{
					Position: eval.Position{
						Filename: "for.acorn",
						Offset:   42,
						Line:     4,
						Column:   19,
					},
					Fields: []eval.Field{&eval.KeyValue{
						Key: eval.FieldKey{
							Key: eval.Value{Value: value.String("b")},
							Pos: eval.Position{
								Filename: "for.acorn",
								Offset:   45,
								Line:     5,
								Column:   2,
							},
						},
						Value: &eval.For{
							Value: "y",
							Collection: &eval.Array{
								Pos: eval.Position{
									Filename: "for.acorn",
									Offset:   57,
									Line:     5,
									Column:   14,
								},
								Items: []eval.Expression{
									eval.Value{
										Value: value.Number("1"),
									},
									eval.Value{Value: value.Number("2")},
								},
							},
							Body: &eval.Struct{
								Position: eval.Position{
									Filename: "for.acorn",
									Offset:   63,
									Line:     5,
									Column:   20,
								},
								Fields: []eval.Field{&eval.Embedded{
									Pos: eval.Position{
										Filename: "for.acorn",
										Offset:   67,
										Line:     6,
										Column:   3,
									},
									Expression: eval.Value{Value: value.Boolean(true)},
								}},
							},
							Merge: true,
							Position: eval.Position{
								Filename: "for.acorn",
								Offset:   52,
								Line:     5,
								Column:   9,
							},
						},
						Pos: eval.Position{
							Filename: "for.acorn",
							Offset:   45,
							Line:     5,
							Column:   2,
						},
					}},
				},
				Merge: true,
				Position: eval.Position{
					Filename: "for.acorn",
					Offset:   28,
					Line:     4,
					Column:   5,
				},
			},
		},
	},
}}
